spec/MoneySpec.php

```diff
@@ -385,6 +385,36 @@ function it_throws_an_exception_when_currency_is_different_during_modulus(Calcul
         $this->shouldThrow(\InvalidArgumentException::class)->duringMod(new Money(self::AMOUNT, new Currency(self::OTHER_CURRENCY)));
     }
 
+    function it_denormalizes_money_from_a_stdclass_object()
+    {
+        $normalized = new \stdClass();
```

Comment:
I guess that it would be nice in the test to show its opposite usage:

`$normalized = json_decode(json_encode($money));`

Reply:
Although I (finally) understand your use case, I believe this is fundamentally different from json serialization.

JSON is the serialization part, this is denormalization (converting an intermediate representation to the actual form). You could use it for YAML or XML serialization as well.

That's why I tried to keep this thing separated from JSON from the very beginning.

It could be mentioned in documentation though.